tokio/src/runtime/time/entry.rs

```diff
@@ -126,7 +126,7 @@ impl StateCell {
     fn when(&self) -> Option<u64> {
         let cur_state = self.state.load(Ordering::Relaxed);
 
-        if cur_state == u64::MAX {
+        if cur_state >= STATE_MIN_VALUE {
```

Comment:
It seems like you've also included a fix for #5082 here?

Reply:
It seems so, though I discovered the issue just now. During working on this issue I messed up more in this module and cleared up most of them but this line survived. If you think it's better to not mix #5082 here I'll revert it.